test/translator/common_test.cc

```diff
@@ -150,4 +150,27 @@ TEST(Common, ShouldFailBuildAllocationWhenAllocPageFails) {
   EXPECT_EQ(translator::StatusCode::kFailure, status_code);
 }
 
+TEST(Common, SafePointerCastWrite) {
+  uint32_t expected_val = 0x13292022;
+
+  uint32_t write_buffer = 0;
+  uint8_t* buf = reinterpret_cast<uint8_t*>(&write_buffer);
+  translator::Span<uint8_t> span(buf, sizeof(uint32_t));
+
+  uint32_t* val = translator::SafePointerCastWrite<uint32_t>(span);
+  *val = expected_val;
+
+  ASSERT_EQ(expected_val, write_buffer);
+}
+
+TEST(Common, SafePointerCastRead) {
+  uint32_t expected_val = 0x13292022;
+  uint8_t* buf = reinterpret_cast<uint8_t*>(&expected_val);
+  translator::Span<uint8_t> span(buf, sizeof(uint32_t));
+
+  const uint32_t* val = translator::SafePointerCastRead<uint32_t>(span);
+
+  ASSERT_EQ(expected_val, *val);
+}
+
 }  // namespace
```

Comment:
Do you want tests that cover error cases?

Reply:
Good point, I've gone ahead and added tests which cover all error cases